### ISSUE TITLE
Separate attributes for ping and hostalive thresholds

### ISSUE THREAD

**Issue by b33rcity** [Icinga/icinga2#6376]
Currently, `hostalive` and `ping` both use the attributes `ping_wrta` and `ping_crta` to modify their behavior. In cases where one might be running them in parallel, this creates quite an inconvenience, as one or the other check must be replaced with a customized CheckCommand.  

## Expected Behavior

Specifying `vars.ping_wrta` etc. should modify the behavior of `ping`. A different set of attributes should modify the behavior of `hostalive`. `vars.hostalive_wrta` would be an easy alternative. 

## Context

This issue arose for me while setting up some command delegations. I'm running `ping` from local satellites (i.e. European nodes are pinged by a satellite in Europe), but `hostalive` is running on the master. Since I am monitoring a global infrastructure, being able to see that things are reachable from both "home" and from a local area is very useful. However, if I find I must specify ping thresholds for some node, those thresholds are also applied to hostalive and I get a Host Down alert for a host that is very much alive. 

As a more general argument, I propose that `hostalive` and `ping`, while similar, serve different purposes. This is true for any deployment, and is reflected in the very different, default alerting thresholds for the two checks. If they should have different defaults, they should also be overridden by different attributes.

**Comment by dnsmichi** [comment]
Can you please provide a concrete example configuration? I'm not sure I can follow here.

**Comment by b33rcity** [comment]
Per the docs, default wrta for ping is 100ms and for hostalive is 3000ms.

Assume average latency is as follows:
- master <-> target: 300ms
- satellite <-> target: 150ms

```
object Host "example" {
    address = 1.2.3.4
    check_command = "hostalive"
    vars.ping_wrta = "190, 5%"     // Distance to satellite necessitates this threshold for check_ping
    vars.ping_crta = "230, 15%"
}

apply Service "ping" {
    check_command = "ping"
    command_endpoint = if ("europe" in host.groups) {
        "euroSat.tld"
    } else {
        "master.tld"
    assign where host.address
}
```

With this configuration, `ping` is executed by the satellite in Europe and returns OK. `hostalive` is executed by the master and returns DOWN, even though _normally_ `hostalive` would return OK in this situation.

**Comment by dnsmichi** [comment]
In this case, I would set the specific custom attributes in the service object itself.

**Comment by b33rcity** [comment]
This must be a gap in my knowledge, then. If I use `check_command` on a host/host template, that attribute is going directly to the corresponding CheckCommand object, correct? If I define a Service object of the same name as the CheckCommand, it does _not_ get used for `check_command`?

eg

```
template Host "generic-host" {
    check_command = "hostalive"
}

apply Service "hostalive" {
    check_command = "hostalive"
    vars. ....
}

object CheckCommand "hostalive" {
    ....
}
```
Only whatever is defined by the CheckCommand is what will be used by the template and any Host which inherits it? (Barring an `assign where host.address` in the apply, which intuition states would result in the Host object having two seperate Hostalive checks being run against it.)

**Comment by dnsmichi** [comment]
You're mixing things here. The key is to define what your checkable object should do. This can either be a service or a host.

If you say that your host is reachable with ICMP requests, you'd likely want to use ping (which in term is hostalive, but with different default thresholds). On the other hand, if you want to behave it faster, you can also use `icmp`. 

In addition to a slowly answering host you might actually want to check the ping response. This is interesting for network latency, packet loss, etc. and needs different thresholds. If you require your own ping threshold, you just set it inside the service object/apply definition, not in the host object.

TL;DR - this discussion should take place on the community channels where others may share their experience with host icmp checks and service pings. I don't see a change in Icinga required here.

**Comment by Crunsher** [comment]
It's not a bug but we could cut down on the confusion by renaming the variables for the hostalive check. But then there is the issue of an upgrade path...

**Comment by dnsmichi** [comment]
I'm waiting for others to share their opinion. As said, the larger audience is on the forums.

**Comment by b33rcity** [comment]
I ran into this issue again today after months of working around it with `ignore where` statements and custom attributes. To a large degree, actually, that was pretty sufficient. I happened to notice today that one of my hosts running `hostalive` for its check command was alerting at 200ms RTA, which reminded me of this thread :) 

I went ahead and dug into plugin-commands.conf and essentially duplicated the `ping` objects' logic into the `hostalive` objects and removed all links back to `ping`. Now, the only thing `hostalive` and `ping` share with each other is importing the `ipv4-or-ipv6` template, and `hostalive` now takes its arguments from `vars.alive_xyz`. 

Made it look like every host I'm monitoring dropped for a while there :laughing: . In hindsight, I really should have worked on a test platform first... live 'n' learn, eh?

Tomorrow I'll pull the repo so I can send my changes as a PR.

**Comment by htriem** [comment]
Glad you solved your problem. 👍
